packages/account/src/Components/forms/personal-details-form.jsx

```diff
@@ -109,473 +102,592 @@ const PersonalDetailsForm = ({
+                                                onItemSelection={({ value, text }) =>
+                                                    setFieldValue('citizen', value ? text : '', true)
+                                                }
+                                                list_portal_id='modal_root'
+                                                required
+                                                data-testid='citizenship'
+                                            />
+                                        </DesktopWrapper>
+                                        <MobileWrapper>
+                                            <SelectNative
+                                                placeholder={localize('Citizenship')}
+                                                name={field.name}
+                                                disabled={
+                                                    (values?.citizen && is_fully_authenticated) ||
+                                                    isFieldImmutable('citizen', editable_fields) ||
+                                                    (values?.citizen && has_real_account)
+                                                }
+                                                label={is_mf ? localize('Citizenship*') : localize('Citizenship')}
+                                                list_items={residence_list}
+                                                value={values.citizen}
+                                                use_text={true}
+                                                error={touched.citizen && errors.citizen}
+                                                onChange={e => {
+                                                    handleChange(e);
+                                                    setFieldValue('citizen', e.target.value, true);
+                                                }}
+                                                {...field}
+                                                required
+                                                should_hide_disabled_options={false}
+                                                data_testid='citizenship_mobile'
+                                            />
+                                        </MobileWrapper>
+                                    </React.Fragment>
+                                )}
+                            </Field>
+                        )}
+                        {!is_svg && 'phone' in values && (
+                            <PhoneField
+                                value={values.phone}
+                                editable_fields={editable_fields}
+                                has_real_account={has_real_account}
+                                required
+                            />
+                        )}
+                        {!is_svg && ('tax_residence' in values || 'tax_identification_number' in values) && (
```

Comment:
nit: maybe we can check !is_svg for PhoneField and TaxResidenceField in a single block

Reply:
Eventually we'll be refactoring this form next, we'll make it cleaner 👍